src/App.tsx

```diff
@@ -15,35 +16,126 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  DEFAULT,
+  ALPHABETICALLY,
+  LENGTH,
+}
+
+interface SortingConditionals {
+  sortingType: SortType;
+  isReversed: boolean;
+}
+
+const sortingGoodsFunction = (
+  goodsArray: string[],
+  { sortingType, isReversed }: SortingConditionals,
+) => {
+  let goodsArrayCopy = [...goodsArray];
+
+  switch (sortingType) {
+    case SortType.ALPHABETICALLY:
+      goodsArrayCopy = [...goodsArrayCopy].sort((good1, good2) =>
+        good1.localeCompare(good2),
+      );
+      break;
+
+    case SortType.LENGTH:
+      goodsArrayCopy = [...goodsArrayCopy].sort(
+        (good1, good2) => good1.length - good2.length,
+      );
+      break;
+
+    default:
+      break;
+  }
```

Comment:
You've already made a copy in `goodsArrayCopy`, there's no need to make another copy of a copy to sort it.

```suggestion
  switch (sortingType) {
    case SortType.ALPHABETICALLY:
      goodsArrayCopy.sort((good1, good2) => good1.localeCompare(good2));
      break;

    case SortType.LENGTH:
      goodsArrayCopy.sort((good1, good2) => good1.length - good2.length);
      break;

    default:
      break;
  }
```